From ec8543eba37fe6e4008c49814e4f29b32bd8d4fb Mon Sep 17 00:00:00 2001
From: Jon Bernard <jbernard@debian.org>
Date: Tue, 5 Aug 2014 15:59:18 -0400
Subject: [PATCH] Handle missing HOST_NAME_MAX

On FreeBSD, the HOST_NAME_MAX constant is not found in the expected
location and this causes a build failure.  This patch falls back to
using _POSIX_HOST_NAME_MAX, and then simply 256.
---
 configure.ac | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/configure.ac b/configure.ac
index 80eba33..248044e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,6 +50,17 @@ AC_C_INLINE
 AC_TYPE_PID_T
 AC_TYPE_SIZE_T
 
+# FreeBSD does not define HOST_NAME_MAX
+AC_CHECK_DECL(HOST_NAME_MAX,,
+	[
+	 AC_CHECK_DECL(_POSIX_HOST_NAME_MAX,
+		       AC_DEFINE(HOST_NAME_MAX, _POSIX_HOST_NAME_MAX, []),
+		       AC_DEFINE(HOST_NAME_MAX, 256, [Define to 256 if neither have HOST_NAME_MAX nor _POSIX_HOST_NAME_MAX]),
+		       [[#include <limits.h>]])
+	],
+	[[#include <limits.h>]]
+)
+
 # Checks for library functions.
 AC_FUNC_MALLOC
 AC_FUNC_MMAP
-- 
1.9.1

